felix/bpf/conntrack/scanner.go

```diff
@@ -156,14 +167,29 @@ func (s *Scanner) Scan() {
 				"entry": ctVal,
 			}).Debug("Examining conntrack entry")
 		}
+		val = cleanupv1.Value{}
+		if s.ipVersion == 6 {
+			val = cleanupv1.ValueV6{}
+		}
 
 		for _, scanner := range s.scanners {
-			if verdict := scanner.Check(ctKey, ctVal, s.get); verdict == ScanVerdictDelete {
+			if verdict, ts := scanner.Check(ctKey, ctVal, s.get); verdict == ScanVerdictDelete {
 				if debug {
 					log.Debug("Deleting conntrack entry.")
 				}
-				cleaned++
-				return maps.IterDelete
+				revKey := []byte{}
+				if ctVal.Type() == TypeNATForward {
+					revKey = ctVal.ReverseNATKey().AsBytes()
+				}
+
+				val = cleanupv1.NewValue(revKey, uint64(ctVal.LastSeen()), uint64(ts))
+				if s.ipVersion == 6 {
+					val = cleanupv1.NewValueV6(revKey, uint64(ctVal.LastSeen()), uint64(ts))
+				}
+				err := s.ctCleanupMap.Update(ctKey.AsBytes(), val.AsBytes())
```

Comment:
Is this an ordinary map; should we use a caching map so that we can do batch writes (once we add that feature to caching map!)?

If a caching map, it needs to be flushed.